metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/DirectSqlGetPartition.java

```diff
@@ -522,9 +522,20 @@ public int getPartitionCount(
                     populateParameters(serdeIds, SQL.SQL_GET_SERDE_PARAMS,
                         "serde_id", serdeParams)));
             }
+            ListenableFuture<List<Void>> future = null;
             try {
-                Futures.transform(Futures.successfulAsList(futures), Functions.constant(null)).get(1, TimeUnit.HOURS);
-            } catch (Exception e) {
+                future = Futures.allAsList(futures);
+                final int getPartitionsDetailsTimeout = Integer.parseInt(configuration
+                    .getOrDefault(HiveConfigConstants.GET_PARTITION_DETAILS_TIMEOUT, "120"));
+                future.get(getPartitionsDetailsTimeout, TimeUnit.SECONDS);
```

Comment:
How the 120 seconds was picked? If it's too short, we may get more pressure from the applications retries.  Can we get a rough idea based on the current metrics?